unic-langid-impl/src/bin/generate_layout.rs

```diff
@@ -63,68 +57,98 @@ fn main() {
     let path = "./data/cldr-misc-full/main/";
     let map = langid_to_direction_map(path);
 
-    let mut scripts = HashSet::new();
-    let mut langs = HashSet::new();
+    let mut scripts_ltr = HashSet::new();
+    let mut scripts_rtl = HashSet::new();
+    let mut scripts_ttb = HashSet::new();
+    let mut langs_rtl = HashSet::new();
 
     for (langid, dir) in map.iter() {
+        let script = langid.script;
+
+        let scripts;
+        let exp_dir;
         if dir == &CharacterDirection::LTR {
-            continue;
+            scripts = &mut scripts_ltr;
+            exp_dir = &CharacterDirection::LTR;
+        } else if dir == &CharacterDirection::RTL {
+            scripts = &mut scripts_rtl;
+            exp_dir = &CharacterDirection::RTL;
+
+            let lang = langid.language;
+            langs_rtl.insert(lang);
+        } else {
+            scripts = &mut scripts_ttb;
+            exp_dir = &CharacterDirection::TTB;
```

Comment:
`nit`: Rust supports `if/else` as an expression. You can do:

```rust
let (scripts, exp_dir) = if ... {
  (&mut scripts_ltr, CharacterDirection::LTR)
} else {
  (&mut scripts_rtl, CharacterDirection::RTL)
};
```

or even better:

```rust
let (scripts, exp_dir) = match dir {
  CharacterDirection::LTR => (&mut scripts_ltr, CharacterDirecton::LTR),
  CharacterDirection::RTL => {
      langs_rtl.insert(langid.language);
      (&mut scripts_rtl, CharacterDirection::RTL)
  },
  ...
}
```

Reply:
Done. Actually, the `exp_dir` turned out to be unnecessary and got factored out.